replication_state.go

```diff
@@ -296,9 +326,41 @@ func (r *ReplicationState) GetBlockWithSeq(seq uint64) Block {
 	return nil
 }
 
-func (r *ReplicationState) requestDigests(digests []Digest) {
-	// TODO: In a future PR, I will add a message that requests a specific digest.
-	r.logger.Debug("Not implemented yet", zap.Stringers("Digests", digests))
+func (r *ReplicationState) requestDigests(missingBlocks []seqAndDigest) {
+	// grab the lock since this is called in the timeout handler goroutine
+	r.epochLock.Lock()
+	defer r.epochLock.Unlock()
+
+	signedQuorum := r.roundRequestor.getHighestObserved()
+	if signedQuorum == nil {
+		signedQuorum = r.finalizationRequestor.getHighestObserved()
+	}
+	if signedQuorum == nil {
+		r.logger.Warn("Replication State cannot request missing block digests, no known nodes to request from")
```

Comment:
we call `requestDigests` because we called `processNotarizedBlock`. 

Can we do this?

```
	e.replicationState.roundRequestor.observedSignedQuorum(&signedQuorum{
		round: notarization.Vote.Round,
		signers: notarization.QC.Signers(),
	}, notarization.Vote.Round)
	e.blockVerificationScheduler.ScheduleTaskWithDependencies(task, md.Seq, blockDependency, missingRounds)
```

Won't that eliminate the chances the `==` check will fail?